macros/blueprint-proc-macro/src/job.rs

```diff
@@ -479,7 +461,7 @@ pub(crate) fn get_event_handler_args<'a>(
     let event_handler_args = x.difference(&y).copied().collect::<Vec<_>>();
     let event_handler_types = event_handler_args
         .iter()
-        .map(|r| param_types.get(*r).unwrap())
+        .map(|r| param_types.get(*r).expect("Should exist"))
         .collect::<Vec<_>>();
```

Comment:
Map to proper errors instead of expect/panic here then collect into a `Result<Vec<_>>()` for proper error handling in this case.